### PR TITLE
Using new API version 2022-03-01 and adding new Reserved Resource types support for getCatalog

### PR DESCRIPTION
**Pull request by gaoyp830**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by archerzz — CHANGES_REQUESTED** [review]
@gaoyp830 Please update the following files according to [release proccess](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process):
- https://github.com/Azure/azure-sdk-for-net/blob/main/eng/mgmt/mgmtmetadata/reservations_resource-manager.txt
- https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/reservations/Microsoft.Azure.Management.Reservations/src/Properties/AssemblyInfo.cs

Thanks

**Comment by gaoyp830** [comment]
> * [src/Properties/AssemblyInfo.cs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/reservations/Microsoft.Azure.Management.Reservations/src/Properties/AssemblyInfo.cs)

Ran generate.ps1 and regenerated some new changes. Updated both reservations_resource-manager.txt and AssemblyInfo.cs

File: sdk/reservations/Microsoft.Azure.Management.Reservations/src/Generated/Models/ReservationResponse.cs [hunk 1]
```diff
@@ -75,14 +86,16 @@ public ReservationResponse(string location = default(string), int? etag = defaul
...
+        /// Gets or sets the sku information associated to this reservation
         /// </summary>
         [JsonProperty(PropertyName = "sku")]
         public SkuName Sku { get; set; }
 
         /// <summary>
+        /// Gets or sets the properties associated to this reservation
         /// </summary>
         [JsonProperty(PropertyName = "properties")]
-        public ReservationProperties Properties { get; set; }
```
Comment (archerzz) [inline sdk/reservations/Microsoft.Azure.Management.Reservations/src/Generated/Models/ReservationResponse.cs:85]: breaking change

**Review by archerzz — CHANGES_REQUESTED** [review]
There are some breaking changes. And it seems that we cannot add work-around for backward compatibility.
Please bump `AssemblyVersion` and `AssemblyFileVersion` to the next major version (e.g. `2.0.0.0`)

**Comment by gaoyp830** [comment]
> There are some breaking changes. And it seems that we cannot add work-around for backward compatibility. Please bump `AssemblyVersion` and `AssemblyFileVersion` to the next major version (e.g. `2.0.0.0`)

Updated to new major version 2.0.0

**Review by archerzz — APPROVED** [review]
@gaoyp830 I just realized that the SDK seems still in `preview` release. Can you confirm? If so, you can just bump the minor version (e.g. `1.16.0`).

I pre-approve. Please feel free to change the number to what you think is appropriate. I'll merge then.

**Review by archerzz — APPROVED** [review]

**Pull request merged**